schema/record.go

```diff
@@ -293,6 +305,14 @@ func (r *RecordDefinition) AddStruct(p *generator.Package, containers bool) erro
 
 func (r *RecordDefinition) AddSerializer(p *generator.Package) {
 	// Import guard, to avoid circular dependencies
+	if !p.HasFunction(UTIL_FILE, "", GENERIC_DESERIALIZER_FUNC) {
```

Comment:
There's no risk of a circular dependency here - that guard exists on the other block because `r.fields` might contain a type that references this type again (think of a linked list) and we don't want to get into an infinite loop.

Reply:
You're right, let me fix that unnecessary if there...